source/extensions/filters/http/ext_authz/ext_authz.h

```diff
@@ -269,6 +275,7 @@ class Filter : public Logger::Loggable<Logger::Id::filter>,
   bool buffer_data_{};
   bool skip_check_{false};
   envoy::service::auth::v3::CheckRequest check_request_{};
+  mutable const FilterConfigPerRoute* per_route_config_{nullptr};
```

Comment:
It seems odd that this is mutable. Why does it need to be mutable?

Also, technically, I think it's possible for this to be unsafe if you store a raw pointer like this. I think you need to also snap the RoutePtr and hold the shared_ptr to make sure the per route config doesn't go away underneath you?

Reply:
Yes, this is not necessary. I introduced a struct instead when getting the specific config, and hold those flags to the existing (member) booleans.